tvtk/pyface/ui/wx/scene.py

```diff
@@ -658,21 +658,23 @@ def _show_parent_hack(window, parent):
         # messed up when the application window is shown.  To work
         # around this a dynamic IDLE event handler is added and
         # immediately removed once it executes.  This event handler
-        # simply forces a resize to occur.  The _idle_count allows us
-        # to execute the idle function a few times (this seems to work
-        # better).
+        # simply forces a resize to occur.  Previously this event
+        # handler is excecuted for the first two idle events.
+        # However this has caused scene.save to effectively ignore
+        # the `size` attribute on Mac OS (i.e. scene.save resizes the
+        # render window to the target size but then the idle handler
+        # resize it back to the old size before an image is saved.
+        # Since resizing once seems to be efficient, the idle handler
+        # is only called for the first idle event.
         def _do_idle(event, window=window):
             w = wx.GetTopLevelParent(window)
             # Force a resize
             sz = w.GetSize()
             w.SetSize((sz[0]-1, sz[1]-1))
             w.SetSize(sz)
-            window._idle_count -= 1
-            if window._idle_count < 1:
-                wx.EVT_IDLE(window, None)
-                del window._idle_count
+            # Remove the handler
+            wx.EVT_IDLE(window, None)
 
-        window._idle_count = 2
```

Comment:
This change regarding the idle event handler is for fixing the saved image size on Mac OS with wxPython 2.9.2.4 and 3.0.2.0 (tested on OS X 10.11.2).

I don't know why it was better to call the resize function for the first *two* Idle events and not just the first one idle event.  Having tested on Ubuntu and MacOS with wxPython 2.8+, I have not encountered a situation when doing the resize once is insufficient.  Was this for some older wx versions?

Nevertheless, given that `mlab.savefig(size=size)` is the next thing to do after creating a scene, doing the resize for the second idle event results in an image of the wrong size on Mac OS.  This is caused by the idle event handler being called after `scene.set_size` and before the `scene.save_png` (or save_bmp... etc) function inside `scene.save`; i.e. on Mac OS, the event handlers are fast and efficient - nothing wrong on wx part.  For some reason the event handling is slower on Ubuntu (e.g. Travis) and so the size of the saved image was fine.   My OS X version is also recent and the Ubuntu is running on a VM with fewer CPU cores - this could be why.